### ISSUE TITLE
Extending comparisonAlwaysTrue to match comparisonAlwaysFalse in case with Literal

### ISSUE THREAD

**Issue by Gobot1234** [microsoft/pyright#4107]
**Is your feature request related to a problem? Please describe.**
Currently `comparisonAlwaysTrue` does not have feature parity with `comparisonAlwaysFalse` in cases with Literals.

![image](https://user-images.githubusercontent.com/50501825/198751154-a6813130-ef38-4d1d-bc8c-bffa321bcc88.png)
Code
```py
x = 1
if x == 1:
    if x == 1:
        ...
    if x == 2:
        ...

match x:
    case 1:
        if x == 1:
            ...
        reveal_type("always" if x == 2 else "never")
```

**Describe the solution you'd like**
Ideally a warning that the 2nd, 3rd and 4th `if x == 1` saying the condition will always evaluate to True.

**Additional context**
I recently discovered this setting and it would have helped me spot a bug in
```py
if id.type == Group:
    foo("group" if id.type == Group else "user")
```

**Comment by erictraut** [comment]
Thanks for the suggestion. This will be included in the next release.

**Comment by erictraut** [comment]
This change is resulting in a bunch of undesirable noise in internal Microsoft code bases, so I think we'll need to revert it and think about the right approach.

There is a common pattern that shows up frequently in code:
```python
class Color(Enum):
    Red = 0
    Blue = 2

def print_color(color: Color):
    if color == Color.Red:
        print("red")
    elif color == Color.Blue:
        print("blue")
    else:
        raise ValueError("Unknown color")
```

With this change in place, the expression `color == Color.Blue` is flagged as an unnecessary conditional expression.

**Comment by Gobot1234** [comment]
Would it not be possible to suggest using a match statement if you want exhaustiveness checking?

**Comment by erictraut** [comment]
The match statement is available only in Python 3.10 and newer, so that's not an option in many cases (at least not for the next ~two years). Also, that would require changing a bunch of perfectly legit code. The whole purpose of the `reportUnnecessaryComparison` check is to detect likely bugs. If it generates false positives for a bunch of legitimate code, then it loses most of its value.

**Comment by erictraut** [comment]
After thinking about this more, I don't think we can justify making this change. It will generate too much noise (false positives) for legitimate code.

**Comment by Gobot1234** [comment]
The original code snippet will raise an error though now right?

**Comment by erictraut** [comment]
The name of the check is `reportUnnecessaryComparison`. It's designed to highlight comparisons that smell like potential bugs. As such, it doesn't report errors for all comparisons that are provably always True or always False, only those that are likely to point to a bug in the code. If it generates false positives for legitimate code patterns, then it loses much of its utility. I've reverted the recent changes, and it now behaves as it did previously.

**Comment by Gobot1234** [comment]
> The match statement is available only in Python 3.10 and newer, so that's not an option in many cases (at least not for the next ~two years)

Is it worth looking at this again now that 2 years have passed

**Comment by erictraut** [comment]
I think there's still significant potential for false positives here. And in the intervening two years, no one else has requested this change or upvoted this issue. Without stronger signal from other pyright users that this is important, I don't think it's worth the risk of introducing a bunch of noise in the form of false positives.
